dxf_api/src/ipf/live/InstrumentProfileConnection.cs

```diff
@@ -273,22 +275,23 @@ public override string ToString()
         /// </summary>
         private void Handler()
         {
-            DateTime downloadStart = DateTime.MinValue;
+            var downloadStart = DateTime.MinValue;
+            
             while (CurrentState != State.Closed)
             {
                 // wait before retrying
-                if (DateTime.Now.Subtract(downloadStart).TotalMilliseconds > UpdatePeriod)
+                if (!(DateTime.Now.Subtract(downloadStart).TotalMilliseconds > UpdatePeriod)) continue;
+                
+                try
                 {
-                    try
-                    {
-                        Download();
-                    }
-                    catch (Exception e)
-                    {
-                        CallOnError(e);
-                    }
-                    downloadStart = DateTime.Now;
+                    Download();
+                }
+                catch (Exception e)
+                {
+                    CallOnError(e);
                 }
+                
```

Comment:
Do we need to sleep here like in java code?

Reply:
Done